src/logic/map_objects/descriptions.cc

```diff
@@ -0,0 +1,537 @@
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_critter_index(const std::string& critername) const {
+	DescriptionIndex const result = critter_index(critername);
+
+	if (result == INVALID_INDEX) {
+		throw GameDataError("Unknown critter '%s'", critername.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_immovable_index(const std::string& immovablename) const {
+	const DescriptionIndex result =
+	   immovable_index(legacy_lookup_table_->lookup_immovable(immovablename));
+	if (!immovable_exists(result)) {
+		throw GameDataError("Unknown immovable '%s'", immovablename.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_resource_index(const std::string& resourcename) const {
+	DescriptionIndex const result = resource_index(resourcename);
+
+	if (result == INVALID_INDEX) {
+		throw GameDataError("Unknown resource '%s'", resourcename.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_ship_index(const std::string& shipname) const {
+	const DescriptionIndex result = ship_index(legacy_lookup_table_->lookup_ship(shipname));
+	if (!ship_exists(result)) {
+		throw GameDataError("Unknown ship '%s'", shipname.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_terrain_index(const std::string& terrainname) const {
+	DescriptionIndex const result = terrain_index(terrainname);
+
+	if (result == INVALID_INDEX) {
+		throw GameDataError("Unknown terrain '%s'", terrainname.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_tribe_index(const std::string& tribename) const {
+	const DescriptionIndex result = tribe_index(tribename);
+	if (!tribe_exists(result)) {
+		throw GameDataError("Unknown tribe '%s'", tribename.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_ware_index(const std::string& warename) const {
+	const DescriptionIndex result = ware_index(legacy_lookup_table_->lookup_ware(warename));
+	if (!ware_exists(result)) {
+		throw GameDataError("Unknown ware '%s'", warename.c_str());
+	}
+	return result;
+}
+DescriptionIndex Descriptions::safe_worker_index(const std::string& workername) const {
+	const DescriptionIndex result = worker_index(legacy_lookup_table_->lookup_worker(workername));
+	if (!worker_exists(result)) {
+		throw GameDataError("Unknown worker '%s'", workername.c_str());
+	}
+	return result;
+}
+
+DescriptionIndex Descriptions::building_index(const std::string& buildingname) const {
+	return buildings_->get_index(buildingname);
+}
+DescriptionIndex Descriptions::critter_index(const std::string& crittername) const {
+	return critters_->get_index(crittername);
+}
+DescriptionIndex Descriptions::immovable_index(const std::string& immovablename) const {
+	return immovables_->get_index(immovablename);
+}
+DescriptionIndex Descriptions::resource_index(const std::string& resourcename) const {
+	return resourcename != "none" ? resources_->get_index(resourcename) : Widelands::kNoResource;
```

Comment:
Is this special value actually needed and used? Is it documented somewhere?

Reply:
I agree that this looks like code smell. I'll add a comment for now - changing it might break stuff.